subsys/settings/src/settings_runtime.c

```diff
@@ -22,7 +22,7 @@ int settings_runtime_set(const char *name, void *data, size_t len)
 	char *name_argv[SETTINGS_MAX_DIR_DEPTH];
 	int name_argc;
 
-	strncpy(name1, name, sizeof(name1));
+	strncpy(name1, name, sizeof(name1) - 1);
```

Comment:
wouldn't you need to zero the last byte after?

Reply:
Right! :)

Reply:
> The strncpy() function shall copy no more than n characters from the null-terminated string pointed to by src to the memory pointed to by dest. The destination shall be padded with null characters.

> Beware: A terminating null byte will not be written to the resulting dest string when the length of the src string is greater than or equal to n (this follows from the above description, it's not a new constraint).

Reply:
bad fix